src/Domain/Utils/FileHandlerTrait.php

```diff
@@ -4,41 +4,44 @@
 
 namespace CubaDevOps\Flexi\Domain\Utils;
 
+use RuntimeException;
+
 trait FileHandlerTrait
 {
     use OSDetector;
 
-    public function writeToFile(string $file_path, string $content, int $flags = 0): void
+    /**
+     * @param string $file_path
+     * @param string $content
+     * @param int $flags
+     * @param bool $try_create_it
+     * @return void
+     * @throws RuntimeException
+     */
+    public function writeToFile(string $file_path, string $content, int $flags = 0, bool $try_create_it = false): void
     {
-        $this->ensureFileExists($file_path);
+        $this->ensureFileExists($file_path, $try_create_it);
```

Comment:
Is is better if you have a separate function for fileExists($path): bool) -> if(result === false && $trytoCreateit) $this->createFile();

Reply:
It is a great idea to have a separate function to check if the file exists and to improve the semantic naming of functions by following the "tell, don't ask" principle.